components/places/ios/Places/Places.swift

```diff
@@ -0,0 +1,418 @@
+     *
+     * - Returns: The GUID of the newly inserted bookmark separator.
+     */
+    func createSeparator(parentGUID: String, position: UInt32? = nil) throws -> String {
+        return try queue.sync {
+            try self.checkApi()
+            let msg = insertionMsg(type: .separator, parentGUID: parentGUID, position: position)
+            return try doInsert(msg: msg)
+        }
+    }
+
+    /**
+     * Create a bookmark item, returning it's guid.
+     *
+     * - Parameter parentGUID: The GUID of the (soon to be) parent of this bookmark.
+     *
+     * - Parameter position: The index where to insert the record inside
+     *                       it's parent. If not provided, this item will
+     *                       be appended.
+     *
+     * - Parameter url: The URL to bookmark
+     *
+     * - Parameter title: The title of the new bookmark, if any.
+     *
+     * - Returns: The GUID of the newly inserted bookmark item.
+     */
+    func createBookmark(parentGUID: String, url: String, title: String?, position: UInt32? = nil) throws -> String {
+        return try queue.sync {
+            try self.checkApi()
+            var msg = insertionMsg(type: .bookmark, parentGUID: parentGUID, position: position)
+            msg.url = url
+            if let t = title {
+                msg.title = t
+            }
+            return try doInsert(msg: msg)
+        }
+    }
+
+    /**
+     * Update a bookmark to the provided info.
+     *
+     * - Parameters:
+     *     - guid: Guid of the bookmark to update
+     *
+     *     - parentGUID: If the record should be moved to another folder, the guid
+     *                   of the folder it should be moved to. Interacts with
+     *                   `position`, see the note below for details.
+     *
+     *     - position: If the record should be moved, the 0-based index where it
+     *                 should be moved to. Interacts with `parentGUID`, see the note
+     *                 below for details
+     *
+     *     - title: If the record is a `BookmarkNodeType.bookmark` or a `BookmarkNodeType.folder`,
+     *              and it's title should be changed, then the new value of the title.
+     *
+     *     - url: If the record is a `BookmarkNodeType.bookmark` node, and it's `url`
+     *            should be changed, then the new value for the url.
+     *
+     * - Note: The `parentGUID` and `position` parameters interact with eachother
+     *   as follows:
+     *
+     *     - If `parentGUID` is not provided and `position` is, we treat this
+     *       a move within the same folder.
+     *
+     *     - If `parentGUID` and `position` are both provided, we treat this as
+     *       a move to / within that folder, and we insert at the requested
+     *       position.
+     *
+     *     - If `position` is not provided (and `parentGUID` is) then it's
+     *       treated as a move the end of that folder.
+     */
+    func updateBookmark(guid: String,
```

Comment:
nit: maybe should be named `updateBookmarkNode` since it can operate on bookmarks, folders and separators?